images/linux/scripts/installers/haskell.sh

```diff
@@ -14,29 +14,29 @@ export BOOTSTRAP_HASKELL_GHC_VERSION=0
 ghcup_bin=$GHCUP_INSTALL_BASE_PREFIX/.ghcup/bin
 setEtcEnvironmentVariable "BOOTSTRAP_HASKELL_NONINTERACTIVE" $BOOTSTRAP_HASKELL_NONINTERACTIVE
 
-# Install Herbert V. Riedel's PPA for managing multiple version of ghc on ubuntu.
-# https://launchpad.net/~hvr/+archive/ubuntu/ghc
-apt-get install -y software-properties-common
-add-apt-repository -y ppa:hvr/ghc
-apt-get update
-
 # Install GHCup
 curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh > /dev/null 2>&1 || true
 export PATH="$ghcup_bin:$PATH"
 prependEtcEnvironmentPath $ghcup_bin
 
-# Get 2 latest Haskell Major.Minor versions
-allGhcVersions=$(apt-cache search "^ghc-" | grep -Po '(\d*\.){2}\d*' | sort --unique --version-sort)
-ghcMajorMinorVersions=$(echo "$allGhcVersions" | cut -d "." -f 1,2 | sort --unique --version-sort | tail -2)
-
-# We are using apt-get to install ghc, not ghcup,
-# because ghc installed through ghcup takes up too much disk space (2GB versus 1GB through apt-get)
-for version in $ghcMajorMinorVersions; do
-    # Get latest patch version for given Major.Minor one (ex. 8.6.5 for 8.6) and install it
-    exactVersion=$(echo "$allGhcVersions" | grep $version | sort --unique --version-sort | tail -1)
-    apt-get install -y ghc-$exactVersion
-    ghcInstalledVersions+=("$exactVersion")
-    defaultGHCVersion=$exactVersion
+availableVersions=$(ghcup list -t ghc -r | grep -v "prerelease" | awk '{print $2}')
+
+# Install 3 latest major versions(For instance 8.6.5, 8.8.4, 8.10.2)
+minorMajorVersions=$(echo "$availableVersions" | cut -d"." -f 1,2 | uniq | tail -n3)
```

Comment:
@hasufell , Currently, we install the last 2 major version. Could you please remove one?

```
allGhcVersions=$(apt-cache search "^ghc-" | grep -Po '(\d*\.){2}\d*' | sort --unique --version-sort)
echo "$allGhcVersions" | cut -d "." -f 1,2 | sort --unique --version-sort | tail -2
8.10
9.0
```